src/cop0.rs

```diff
@@ -586,10 +587,13 @@ pub unsafe fn cache64<const OP: u8, const OFFSET: u16>(location: u64) {
 // Fires a cop2 usable exception and handles it itself. Afterwards, Cause.copindex will be set to 2
 pub fn preset_cause_to_copindex2() -> Result<(), String> {
     // Fire a COP2 unusable exception. This is to write something into Cause.copindex so that we can see whether it gets overwritten
-    let temp_context = expect_exception(CauseException::CopUnusable, 1, || {
+    let temp_context = match expect_exception(CauseException::CopUnusable, 1, || {
         unsafe { asm!("MFC2 $0, $0"); }
         Ok(())
-    })?;
+    }) {
+        Ok(c) => c,
+        Err(e) => return Err(format!("Failed to preset Cause.copindex to 2: {}", e))
+    };
     if temp_context.cause != Cause::new().with_coprocessor_error(u2::new(2)).with_exception(CauseException::CopUnusable) {
         return Err("Presetting Cause.copindex to 2 failed".to_string());
```

Comment:
Might be worth changing this line, otherwise it can be equally confusing.

"Cause.copindex was supposed to be preset to 2, but it didn't change"

Reply:
Done